extension/natives/eventresponder.h

```diff
@@ -25,4 +25,287 @@ EVENTRESPONDERNATIVE(NextContainedResponder)
 	return (cell_t)(pResponder->NextContainedResponder(pPrevResponder));
 }
 
+EVENTRESPONDERNATIVE(OnLeaveGround)
+	CBaseEntity* entity = gamehelpers->ReferenceToEntity(params[2]);
+	if (!entity && params[2] != -1)
+	{
+		return pContext->ThrowNativeError("Entity %d is invalid!", params[2]);
+	}
+	pResponder->OnLeaveGround(entity);
+	return 0;
+}
+
+EVENTRESPONDERNATIVE(OnLandOnGround)
+	CBaseEntity* entity = gamehelpers->ReferenceToEntity(params[2]);
+	if (!entity && params[2] != -1)
+	{
+		return pContext->ThrowNativeError("Entity %d is invalid!", params[2]);
+	}
+	pResponder->OnLandOnGround(entity);
+	return 0;
+}
+
+// TO-DO: OnContact
+
+EVENTRESPONDERNATIVE(OnMoveToSuccess)
+	pResponder->OnMoveToSuccess((Path*)params[2]);
```

Comment:
```suggestion
	Path* path = (Path*)params[2];
	if (!path)
	{
		return pContext->ThrowNativeError("Path is NULL");
	}

	pResponder->OnMoveToSuccess(path);
```

Reply:
Intentional, I was wondering, maybe plugin makers & their custom intention will want to pass null path.